File: C++/Koin.cpp
#include "Koin.hpp"

const double KSpeed = 1;        // kecepatan koin
const int minNilaiKoin = 15;   // harga minimum koin
const int maxNilaiKoin = 30;   // harga maksimum koin

Koin::Koin(int newNilaiKoin, Point newPosisi) : ObjekMati(KSpeed){
  nilaiKoin = newNilaiKoin;
  jenis = "Koin";
  ObjekMati::setPosisi(newPosisi);
}

//getter
int Koin::getNilaiKoin(){
  return nilaiKoin;
}

void Koin::setNilaiKoin(int newNilaiKoin){
  nilaiKoin = newNilaiKoin;
}

File: Java/Koin.java
public class Koin extends ObjekMati {

  private final int kecepatanKoin = 1;
  private int nilaiKoin;

  public Koin(int nilaiKoin, Point posisi) {
    super(1);
    this.nilaiKoin = nilaiKoin;
    this.jenis = "Koin";
    this.posisi = posisi;
  }

  public int getNilaiKoin() {
    return nilaiKoin;
  }

  public void setNilaiKoin(int nilaiKoin) {
    this.nilaiKoin = nilaiKoin;
  }
}
